source/server/listener_manager_impl.h

```diff
@@ -36,9 +36,11 @@ class ProdListenerComponentFactory : public ListenerComponentFactory,
   }
   Network::ListenSocketPtr createListenSocket(Network::Address::InstanceConstSharedPtr address,
                                               bool bind_to_port) override;
+  uint64_t nextOpaqueId() override { return next_opaque_id_++; }
 
 private:
   Instance& server_;
+  uint64_t next_opaque_id_{};
```

Comment:
This will start from 0 but the opaque ID of 0 is also used as a special case for the admin listener.  Though I think that'll always be in a different thread's ConnectionHandler this might be confusing.  Should we start from 1 here?